src/routes/PublicRoute.js

```diff
@@ -0,0 +1,12 @@
+import React from "react";
```

Comment:
How are you planning to use the Plublic Route?
just `<Route .... />` should be enough if the route is not auth protected.

Reply:
I was going to use PublicRoute and Protected Route, i have tried to import and check the isLoggedIn from the context inside ProtectedRoute but somehow it's not getting the last state for isLoggedIn after validate the token, so for now i don't use it instead giving the condition inside switch route

Reply:
`export const validateByToken = (token) =>{
    return async (dispatch) => {
        // you can dispatch reducer here
        const sendRequest = async () =>{
            const response = await fetch('https://blog-app-react-fb813-default-rtdb.asia-southeast1.firebasedatabase.app/admin.json',
            {
                method : 'GET',
            }
            );
            if(!response.ok){
                throw new Error('Getting admin data failed!')
            }

            const data = response.json()
            return data
        }

        try {
           const responseData =  await sendRequest();
           const adminData = responseData.find( u => u.token === token)
           if(adminData){
                dispatch(authAction.authenticate({message:'Success', isAuthenticated : true, token : adminData.token}))
                console.log('Success validate');
           }else{
                dispatch(authAction.authenticate({message:'Token is not valid', isAuthenticated : false}))
           }
        } catch (error) {
            console.log(error.message);
        }
    }
}`

i have this http async action creators to validate the token and dispatch the state for auth. The auth context, will get the token from local storage and dispatch the validate action on componentDidMount. 

after the state is updated, the auth context also listen the state updates. if isAuthenticated state is true then save the token and set the local state isLoggedIn ( auth-context ) into true